packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -17,6 +25,85 @@ export interface RemotePlugin {
   settings: JSONObject
 }
 
+class ActionDestination implements Plugin {
+  name: string // destination name
+  version = '1.0.0'
+  type: Plugin['type']
+
+  alternativeNames: string[] = []
+
+  middleware: DestinationMiddlewareFunction[] = []
+
+  action: Plugin
+
+  constructor(name: string, action: Plugin) {
+    this.action = action
+    this.name = name
+    this.type = action.type
+    this.alternativeNames.push(action.name)
+  }
+
+  addMiddleware(...fn: DestinationMiddlewareFunction[]): void {
+    this.middleware.push(...fn)
+  }
+
+  private async transform(ctx: Context): Promise<Context> {
+    const modifiedEvent = await applyDestinationMiddleware(
+      this.name,
+      klona(ctx.event),
+      this.middleware
+    )
+
+    if (modifiedEvent === null) {
+      ctx.cancel(
+        new ContextCancelation({
+          retry: false,
+          reason: 'dropped by destination middleware',
+        })
+      )
+    }
+
+    return new Context(modifiedEvent!)
+  }
+
+  private _createMethod(
+    methodName: 'track' | 'page' | 'identify' | 'alias' | 'group' | 'screen'
+  ) {
+    return async (ctx: Context): Promise<Context> => {
+      if (!this.action[methodName]) return ctx
+
+      const transformedContext = await this.transform(ctx)
+      await this.action[methodName]!(transformedContext)
+
+      return ctx
+    }
+  }
+
+  alias = this._createMethod('alias')
+  group = this._createMethod('group')
+  identify = this._createMethod('identify')
+  page = this._createMethod('page')
+  screen = this._createMethod('screen')
+  track = this._createMethod('track')
+
+  /* --- PASSTHROUGH METHODS --- */
+  isLoaded(): boolean {
+    return this.action.isLoaded()
+  }
+
+  ready(): Promise<unknown> {
+    return this.action.ready ? this.action.ready() : Promise.resolve()
+  }
+
+  load(ctx: Context, analytics: Analytics): Promise<unknown> {
+    return this.action.load(ctx, analytics)
+  }
+
+  unload(ctx: Context, analytics: Analytics): Promise<unknown> | unknown {
+    return this.action.unload && this.action.unload(ctx, analytics)
```

Comment:
just in case you aren’t a Typescript nerd, you can shorten this to “return this.action.unload?.(ctx, analytics)” IIRC